### PR TITLE
Fix migrations

### PR DESCRIPTION
**Pull request by s-taylor**
This PR primarily updates redbeard's migrations to be in sync with the new version of redbeard.
There are two scripts available...

* migration - which is solely used to create migration files and associated tests (it does not interact with the database)
* migrate - which is used for running, rollling back, listing migrations and also mutex related commands (all commands interact with the database in some way). 

 A brief summary of changes...
1) Make migration scripts executable within the /bin folder.
2) Document how to use the new migration scripts
3) Use thinky to establish the _migraitons table as well as well as for all queries
4) Use generators in all the places!

See issue https://github.com/Prismatik/redbeard/issues/154

### CODE REVIEW

**Comment by MadRabbit** [comment]
firstly, great work! this looks really good to me.

but a few things  i think need to be addressed

1. `npm dockertest` explodes at eslint checks. very minor i guess
2. do you reckon we could inline the `src/utils/templates` into `src/utils/migration`? maybe during the `redbeard migration` command or something?

i can see the merit, but it seems that for something as simple as this there is no point in dragging all the code generation tools into the actual application.

another option (and maybe a better one) would be to have subcommands to `redbeard migration`, like for example `redbeard migration init` sets up the migration structures and utils; and then `redbeard migration create` will create a singular migration file/test.

basically the idea, i want to avoid adding to apps anything that is not directly related to the application specific logic, and code generators seems a bit off the beam here.

do you reckon you could make the changes today?

**Comment by s-taylor** [comment]
Over to you really, I can do either option. So we have two options...
1. inline the templates into the migration.js file.
2. keep the creation of migrations a redbeard only thing that you just run each time you need a new migration.

The problem I see with option 2. is it's kinda hard to keep your project redbeard and redbeard itself in sync version wise. Like if we change migrations tomorrow so that it now has a run and rollback command (instead of up and down), and you update redbeard, the migrations it now generates might no longer work for your project. So I think inline-ing is probably safer?

**Comment by s-taylor** [comment]
A third option would be to publish the migration creation as an npm package and require it in, but I feel like that's kind against the ethos of redbeard where you have the power to change anything.

**Comment by s-taylor** [comment]
@MadRabbit Should be good to go now, i've inlined the templates into the migrations.js file in utils.
Also dockertests pass on mine, but you might be experiencing the caching issue I keep battling with where you change something but within docker your changes don't come through.

**Comment by MadRabbit** [comment]
ah, perfect, thanks!
